acceptance-tests/cloudsql_mysql_migration_test.go

```diff
@@ -0,0 +1,116 @@
+
+		serviceBroker := brokers.Create(
+			brokers.WithPrefix("csb-mysql"),
+			brokers.WithEnv(apps.EnvVar{Name: brokers.PlansMySQLVar, Value: mysql56plan}),
+		)
+		defer serviceBroker.Delete()
+
+		By("asynchronously starting the target service instance creation")
+		databaseName := random.Name(random.WithPrefix("migrate-database"))
+		targetServiceInstance := services.CreateInstance(
+			"csb-google-mysql",
+			"default",
+			services.WithParameters(map[string]any{
+				"db_name":                    databaseName,
+				"public_ip":                  false,
+				"backups_retain_number":      0,
+				"allow_insecure_connections": true, // MySQL 5.6 does not support TLS bindings
+			}),
+			services.WithBroker(serviceBroker),
+			services.WithAsync(),
+		)
+
+		By("creating the source service instance")
+		const sourceServiceOffering = "google-cloudsql-mysql-vpc"
+		const sourceServicePlan = "default"
+		sourceServiceInstance := services.CreateInstance(
+			sourceServiceOffering,
+			sourceServicePlan,
+			services.WithBrokerName(legacyBrokerName),
+			services.WithParameters(map[string]any{
+				"tier":            legacyDBTier,
+				"private_network": os.Getenv("GCP_PAS_NETWORK"),
+				"database_name":   databaseName,
+				"backups_enabled": "false",
+				"binlog":          "false",
+			}),
+		)
+		defer sourceServiceInstance.Delete()
+
+		By("binding an app to the source service instance")
+		sourceApp := apps.Push(apps.WithApp(apps.MySQL))
+		defer sourceApp.Delete()
+
+		sourceInstanceBinding := sourceServiceInstance.Bind(sourceApp)
+		sourceApp.Start()
+
+		By("adding some data in the source database")
+		key := random.Hexadecimal()
+		value := random.Hexadecimal()
+		sourceApp.PUT(value, "/key-value/%s", key)
+
+		By("waiting for the target service creation to succeed")
+		services.WaitForInstanceCreation(targetServiceInstance.Name)
+		defer targetServiceInstance.Delete()
+
+		By("extracting a service key for the source service instance")
+		credentials := sourceInstanceBinding.Credential()
+		legacyBinding := legacybindings.ExtractLegacyBinding(credentials)
+
+		By("creating a bucket")
+		bucketName := "bucket-" + databaseName
+		gsql.CreateBackupBucket(bucketName)
+		defer gsql.DeleteBucket(bucketName)
+
+		By("performing the backup")
+		backupURI := gsql.CreateBackup(legacyBinding.InstanceName, databaseName, bucketName)
+
+		By("preparing the restore in the target service instance")
+		targetInstanceIaaSName := fmt.Sprintf("csb-mysql-%s", targetServiceInstance.GUID())
+
+		By("restoring the backup onto the target service instance")
+		gsql.RestoreBackup(backupURI, targetInstanceIaaSName, databaseName)
+
+		By("binding an app to the target service instance")
+		targetApp := apps.Push(apps.WithApp(apps.MySQL))
+		defer targetApp.Delete()
+		targetBinding := targetServiceInstance.Bind(targetApp)
```

Comment:
nit-pick (non-blocking): the `sourceInstanceBinding` should be `Unbind` to help managing services accounts

Reply:
Agreed